src/main/java/net/imagej/ops/threshold/rosin/ComputeRosinThreshold.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * #%L
+ * ImageJ software for multidimensional image processing and analysis.
+ * %%
+ * Copyright (C) 2014 - 2016 Board of Regents of the University of
+ * Wisconsin-Madison, University of Konstanz and Brian Northan.
+ * %%
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * 
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ * 
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ * #L%
+ */
+package net.imagej.ops.threshold.rosin;
+
+import net.imagej.ops.Ops;
+import net.imagej.ops.threshold.AbstractComputeThresholdHistogram;
+import net.imglib2.histogram.Histogram1d;
+import net.imglib2.type.numeric.RealType;
+
+import org.scijava.Priority;
+import org.scijava.log.LogService;
+import org.scijava.plugin.Parameter;
+import org.scijava.plugin.Plugin;
+
+/**
+ * Implements Rosin's threshold method. From : Rosin, Paul L. "Unimodal thresholding." Pattern
+ * recognition 34.11 (2001): 2083-2096.
+ *
+ * Implementation inspired/copied from Paul Rosin C implementation :
+ * http://users.cs.cf.ac.uk/Paul.Rosin/resources/unimodal/
+ *
+ * @author Hadrien Mary
+ */
+@Plugin(type = Ops.Threshold.Rosin.class, priority = Priority.HIGH_PRIORITY)
+public class ComputeRosinThreshold<T extends RealType<T>> extends
+        AbstractComputeThresholdHistogram<T> implements Ops.Threshold.Rosin {
+
+        @Parameter
+        private LogService log;
+
+        @Override
+        public long computeBin(final Histogram1d<T> histo) {;
+
+                long[] hist = histo.toLongArray();
```

Comment:
Otsu do that and I don't see any methods to access to the values by the index. So unless you see a solution I would prefer to stick to that.